polling_stations/apps/data_finder/helpers/routing.py

```diff
@@ -18,60 +17,53 @@ class RoutingHelper:
     }
 
     def __init__(self, postcode):
-        self.postcode = Postcode(postcode).without_space
-        self.get_addresses()
-        self.get_councils_from_blacklist()
+        self.postcode = Postcode(postcode)
+        self.addresses = self.get_addresses()
 
     def get_addresses(self):
-        self.addresses = ResidentialAddress.objects.filter(postcode=self.postcode)
-        return self.addresses
+        return Address.objects.uprns_for_postcode(self.postcode.with_space)
 
-    def get_councils_from_blacklist(self):
-        # if this postcode appears in the blacklist table
-        # return a list of any council ids attached to it
-        # if it is not in the table, we will return []
-        blacklist = Blacklist.objects.filter(postcode=self.postcode)
-        self.councils = [row.lad for row in blacklist]
-        return self.councils
+    @property
+    def polling_stations(self):
+        return {address.polling_station_id for address in self.addresses}
 
     @property
     def has_addresses(self):
         return bool(self.addresses)
 
     @property
-    def has_single_address(self):
-        return self.addresses.count == 1
+    def no_stations(self):
+        """Return true if there are addresses, but no polling station information"""
```

Comment:
I don't understand this. Can you explain it more clearly? Are you saying the comment is incorrect? :confused:

Reply:
Sorry, that was a note mainly to myself to check if the comment is correct. It is. 
If there are no addresses then `self.polling_stations == {}` will be true - rather than {""}.